compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -825,25 +825,136 @@ fn should_encode_visibility(def_kind: DefKind) -> bool {
         | DefKind::ForeignTy
         | DefKind::TraitAlias
         | DefKind::AssocTy
+        | DefKind::TyParam
```

Comment:
Why only `TyParam` and not `LifetimeParam` and `ConstParam`?

Reply:
Some spans are only encoded for diagnostics.
Errors related to type parameters may report different diagnostics than errors related to other generic parameters.

It could make sense to drop extern spans for type parameter diagnostics too, to optimize the "good path".